var/spack/repos/builtin/packages/awscli-v2/package.py

```diff
@@ -11,29 +11,33 @@ class AwscliV2(PythonPackage):
 
     homepage = "https://docs.aws.amazon.com/cli"
     url = "https://github.com/aws/aws-cli/archive/refs/tags/2.13.22.tar.gz"
+    list_url = "https://github.com/aws/aws-cli/tags"
 
-    maintainers("climbfuji")
+    maintainers("climbfuji", "teaguesterling")
 
+    version("2.15.53", sha256="a4f5fd4e09b8f2fb3d2049d0610c7b0993f9aafaf427f299439f05643b25eb4b")
     version("2.13.22", sha256="dd731a2ba5973f3219f24c8b332a223a29d959493c8a8e93746d65877d02afc1")
 
-    depends_on("python@3.8:3.11", type=("build", "run"))
-    depends_on("py-flit-core@3.7.1:3.8.0", type=("build"))
-    depends_on("py-colorama@0.2.5:0.4.6", type=("build", "run"))
-    depends_on("py-docutils@0.10:0.19", type=("build", "run"))
-    depends_on("py-cryptography@3.3.2:40.0.1", type=("build", "run"))
-    depends_on("py-ruamel-yaml@0.15:0.17.21", type=("build", "run"))
-    depends_on("py-ruamel-yaml-clib@0.2:0.2.7", type=("build", "run"))
-    depends_on("py-prompt-toolkit@3.0.24:3.0.38", type=("build", "run"))
-    depends_on("py-distro@1.5:1.8", type=("build", "run"))
-    depends_on("py-awscrt@0.16.4:0.16.16", type=("build", "run"))
-    depends_on("py-python-dateutil@2.1:2", type=("build", "run"))
-    depends_on("py-jmespath@0.7.1:1.0", type=("build", "run"))
-    depends_on("py-urllib3@1.25.4:1.26", type=("build", "run"))
+    with default_args(type=("build", "run")):
+        depends_on("python@3.8:")
+        depends_on("py-flit-core@3.7.1:3.8.0")
+        depends_on("py-colorama@0.2.5:0.4.6")
+        depends_on("py-docutils@0.10:0.19")
+        depends_on("py-cryptography@3.3.2:40.0.1")
+        depends_on("py-ruamel-yaml@0.15:0.17.21")
+        depends_on("py-ruamel-yaml-clib@0.2:0.2.7", when="^python@:3.9")
+        depends_on("py-prompt-toolkit@3.0.24:3.0.38")
+        depends_on("py-distro@1.5:1.8")
+        depends_on("py-awscrt@0.16.4:0.16.16", when="@2.13")
+        depends_on("py-awscrt@0.19.18:0.19.19", when="@2.15")
+        depends_on("py-python-dateutil@2.1:2.8.2")
+        depends_on("py-jmespath@0.7.1:1.0")
+        depends_on("py-urllib3@1.25.4:1.26")
 
     variant("examples", default=True, description="Install code examples")
 
-    @run_after("install")
     @when("~examples")
+    @run_after("install")
```

Comment:
Does this change make a difference? If not, why should it be changed?

Reply:
The post_install step was consistently failing for me with a spack error, thus causing the install to fail. Making this change resolved it.

Reply:
Can you change the two lines above with:
```suggestion
    @run_after("install", when="~examples")
```
? The when argument of `run_after` was added exactly to avoid remembering the ordering.

Also, note that right now the `when` is not taken into consideration - `run_after` registers a function for later execution, and since the function is not `when` decorated the `when` part doesn't count.

Reply:
Thanks for providing this suggestion and clarifying the logic!